index.js

```diff
@@ -1,11 +1,277 @@
 const fs = require('fs');
 const path = require('path');
+const dayjs = require('dayjs');
+
+const filters = require('./filters');
+const queries = require('./queries');
 
 module.exports = {
+  extend: '@apostrophecms/piece-type',
   bundle: {
     directory: 'modules',
     modules: getBundleModuleNames()
-  }
+  },
+  options: {
+    label: 'Event',
+    pluralLabel: 'Events',
+    sort: { start: 1 }
+  },
+  columns: {
+    add: {
+      start: {
+        label: 'Start'
+      }
+    }
+  },
+  fields: {
+    add: {
+      startDate: {
+        label: 'Start date',
+        type: 'date',
+        required: true
+      },
+      allDay: {
+        label: 'Is this an all day event?',
+        type: 'boolean',
+        choices: [
+          {
+            label: 'Yes',
+            value: true
+          },
+          {
+            label: 'No',
+            value: false,
+            showFields: ['startTime', 'endTime']
+          }
+        ],
+        def: false
+      },
+      startTime: {
+        label: 'Start time',
+        type: 'time',
+        def: '09:00 AM',
+        required: true,
+        if: {
+          allDay: false
+        }
+      },
+      endTime: {
+        label: 'End time',
+        type: 'time',
+        def: '05:30 PM',
+        required: true,
+        if: {
+          allDay: false
+        }
+      },
+      dateType: {
+        label: 'What type of event is this?',
+        help:
+          'Select if the event is on a single day, consecutive days, or repeats.',
+        type: 'select',
+        choices: [
+          {
+            label: 'Single Day',
+            value: 'single'
+          },
+          {
+            label: 'Consecutive Days',
+            value: 'consecutive'
+          },
+          {
+            label: 'Recurring',
+            value: 'repeat'
+          }
+        ],
+        def: 'single'
+      },
+      endDate: {
+        label: 'End date',
+        type: 'date',
+        if: {
+          dateType: 'consecutive'
+        }
+      },
+      repeatInterval: {
+        label: 'How often does the event repeat?',
+        type: 'select',
+        choices: [
+          {
+            label: 'Every week',
+            value: 'weeks'
+          },
+          {
+            label: 'Every month',
+            value: 'months'
+          }
+        ],
+        if: {
+          dateType: 'repeat'
+        }
+      },
+      repeatCount: {
+        label: 'How many times does it repeat?',
+        type: 'integer',
+        def: 1,
+        if: {
+          dateType: 'repeat'
+        }
+      },
+      description: {
+        type: 'string',
+        label: 'Description',
+        textarea: true,
+        required: true
+      },
+      image: {
```

Comment:
I don't think this should be included in the default schema. A popular one to add project level I'm sure, but not universal.